lib/Dialect/TritonGPU/IR/LinearLayoutConversions.cpp

```diff
@@ -121,9 +121,15 @@ LinearLayout combineCtaCgaWithShape(LinearLayout ctaLayout,
   return ret;
 }
 
+// Make a LinearLayout from swizzled shared layout defined by
+// "vec", "maxPhase", "perPhase" attributes.
+// swizzleBlocks argument controls if swizzling pattern changes
+// between blocks of maxPhase*perPhase rows.
+template <typename SwizzledSharedEncodingType>
 LinearLayout
 sharedToLinearLayoutNoLeadingOffset(ArrayRef<int64_t> shape,
-                                    SwizzledSharedEncodingAttr shared) {
+                                    SwizzledSharedEncodingType shared,
+                                    bool swizzleBlocks) {
```

Comment:
Can we create a new function instead of using parameters and conditions in this one? Think it as conceptually we are creating a new layout that lives in the `amd/` directory. It's also easier for maintenance.